### ISSUE TITLE
Decide supported virtualization providers

### ISSUE THREAD

**Issue by jimmidyson** [kubernetes/minikube#261]
As discussed in #248, we should limit the supported virt providers. Let's decide so we can target embedding the drivers required rather than fork/RPC the plugin binaries as we're doing now.

My suggestions for defaults:
- OS X - xhyve (natively supported with no extra installation reqs)
- Windows - Hyper-V (as long as it's free & easy to set up - no experience with it)
- Linux/*BSD - KVM

We should perhaps also consider virtualbox as it is supported on all above platforms AFAIK & might be a decent fallback which still limits the size of test matrix?

**Comment by dlorenc** [comment]
That list sounds good to me.

For Linux we might also want a "no vm" option, eventually.

**Comment by dlorenc** [comment]
The KVM driver is currently missing a LICENSE. I just opened an issue to track that. We'll need a LICENSE before we can bundle: https://github.com/dhiltgen/docker-machine-kvm/issues/17

**Comment by chuckbutler** [comment]
I see the original author of #220 didn't comment, so I'll chime in here that supporting whatever drivers are already supported by docker-machine would be a solid +1 from me.

Not everyone wants the xhyve virtualization integration (myself being one of them) as its extremely nice to manage VM's with my existing tools. I can confirm that the parallels provider has been stable and a good user experience when testing docker-machine and other infra/container related tech. I get better battery life out of my rig when using Parallels vs VirtualBox, and even xhyve. That alone makes this worth it to me as i'm a road warrior.

I'll gladly donate time/effort to see this to completion.

**Comment by r2d4** [comment]
Closing this: I think we've settled upon

virtualbox
vmwarefusion
kvm
xhyve
hyperv

**Comment by kedare** [comment]
Hmm a support for parallels would probably be very welcomed as this is probably the most used virtualization system on OS X (The best integration with the system and as far as I know the best performances).

Also, it's supported on Docker-machine with an external plugins: https://github.com/Parallels/docker-machine-parallels

**Comment by denji** [comment]
```
$ minikube start --vm-driver=parallels
Starting local Kubernetes cluster...
Downloading Minikube ISO
 36.00 MB / 36.00 MB [==============================================] 100.00% 0s
F1202 15:09:48.153415   57725 cluster.go:411] Unsupported driver: parallels
```

**Comment by ahobson** [comment]
I have patches for parallels support that work well as far as I can tell (the changes are quite minor).  I see that the issue was closed without including parallels in the supported list, but would a PR change your mind?  I've just submitted a CLA.

**Comment by TylerRick** [comment]
Is there a separate issue for tracking the possibility of running kubernetes natively on Linux without the extra VM layer? https://news.ycombinator.com/item?id=12044317 links here but this issue is closed now. 

I've been using https://github.com/kubernetes/kubernetes/blob/master/docs/devel/local-cluster/docker.md to run Kubernetes natively using just containers but that guide now recommends using Minikube, ironically. :)

**Comment by r2d4** [comment]
@TylerRick I don't believe there's an issue made.  Definitely something on our roadmap though.

**Comment by TylerRick** [comment]
Thanks for the update, @r2d4. I do see "Remove hypervisor on Linux systems" on https://github.com/kubernetes/minikube/blob/master/ROADMAP.md now so that's encouraging :+1:

**Comment by denji** [comment]
> Hypervisor.framework

> Generally, machines with an Intel VT-x feature set that includes **Extended Page Tables (EPT)** and **Unrestricted Mode** are supported. You can determine the availability of Hypervisor APIs on a particular machine at runtime with the sysctl(8) command, passing kern.hv_support as an argument.

https://developer.apple.com/reference/hypervisor

**Comment by maxkuzkin** [comment]
Like many other people here i'm using parallels desktop virtualization on OS X. Is there a way to include it's support, like https://github.com/ahobson/docker-machine-parallels?

**Comment by denji** [comment]
@maxkuzkin https://github.com/kubernetes/minikube/blob/d1565d7d962a6824632343946fa40d1c63fedb9f/pkg/minikube/cluster/cluster.go#L398-L411

**Comment by gwicke** [comment]
For those wondering about progress on native container support on Linux, see [this query for "no-vm"](https://github.com/kubernetes/minikube/search?q=no-vm&type=Issues&utf8=%E2%9C%93) in issues.

**Comment by rawkode** [comment]
I can't find any issue for Linux native support. @gwicke, your query doesn't return anything relevant?

**Comment by rawkode** [comment]
Interesting: https://github.com/kubernetes/minikube/commit/a7c2ff3e8429f738f8f353df4da430ab198aff27

**Comment by bennowak** [comment]
+1 for parallels, it'd been a phenomenal find in my workflow, is pretty robust, and does manage battery life very well!  Please add support!

**Comment by csharpe** [comment]
+1 to Parallels support for OSX.  +1 on battery life! Not sure how VMWare Fusion was selected but I believe parallels is the dominant VM product on OSX. 

Happy to help make this happen.

**Comment by benjaminsweetnam247** [comment]
+1 to Parallels support for OSX. battery life and just nicer to deal with.

**Comment by joadr** [comment]
+1 for Parallels! the fork with parallels working is already there and the request is being made by lots of people over 2 years period. This is 2 years of pure bureaucracy.

**Comment by afbjorklund** [comment]
@joadr : did you have an updated PR for the Parallels driver support ?
I was using it myself for virtualization on the Mac, but that was quite some time ago now...

**Comment by afbjorklund** [comment]
Never mind, found it in the linked issues. Basically requesting support for whatever docker-machine supports... Quite similar to our current “generic” discussion. I guess that @tstromberg can help arbitrate, in what drivers are finally allowed to be “supported” by minikube ?
